### PR TITLE
Update command line arguments

### PR DESCRIPTION
**Pull request by thaink**
- Use positional arguments instead of define
- Explicitly require arguments
- Remove test-only parameters: num_inputs and num_outputs

PiperOrigin-RevId: 288266130

### CODE REVIEW

File: cpp/binary/main.cc [hunk 1]
```diff
@@ -30,159 +31,197 @@ namespace mlperf {
...
-          Flag::CreateFlag("delegate", &delegate,
-                           "Delegate to use for inference, if available. "
-                           "Can be one value of {'nnapi', 'gpu', 'none'}."));
-      flag_list.push_back(
-          Flag::CreateFlag("num_inputs", &num_inputs,
-                           "Number of inputs required by the model."));
-      flag_list.push_back(
-          Flag::CreateFlag("num_outputs", &num_outputs,
-                           "Number of outputs produced by the model."));
+      int num_threads = 4;
```
Comment (terryheo) [inline cpp/binary/main.cc]: Isn't 1 better?

File: cpp/binary/main.cc [hunk 2]
```diff
@@ -30,159 +31,197 @@ namespace mlperf {
...
-  IMAGENET = 1,
-  COCO = 2,
+enum class DatasetType {
+  NONE = 0,
+  DUMMY = 1,
+  IMAGENET = 2,
+  COCO = 3,
 };
+
+BackendType Str2BackendType(std::string* name) {
```
Comment (terryheo) [inline cpp/binary/main.cc]: how about "const std::string&" instead of std::string* ?

File: cpp/binary/main.cc [hunk 3]
```diff
@@ -30,159 +31,197 @@ namespace mlperf {
...
+                       Flag::POSITIONAL)};
+  Flags::Parse(&argc, const_cast<const char**>(argv), flag_list);
+  backend_type = Str2BackendType(&backend_name);
+  dataset_type = Str2DatasetType(&dataset_name);
+  if (backend_type == BackendType::NONE || dataset_type == DatasetType::NONE) {
+    LOG(FATAL) << Flags::Usage(command_line, flag_list);
+    return 1;
+  }
+  // Parsed flags should be removed before the next parsing.
+  flag_list.clear();
```
Comment (terryheo) [inline cpp/binary/main.cc]: I prefer to do this just before flag_list.insert()

File: cpp/binary/main.cc (hunk 2 above)
Comment (thaink) [reply cpp/binary/main.cc]: absl::AsciiStrToUpper take pointer and modify it in place. So we have to use pointer or make a copy string.

File: cpp/binary/main.cc (hunk 3 above)
Comment (thaink) [reply cpp/binary/main.cc]: done

File: cpp/binary/main.cc (hunk 1 above)
Comment (thaink) [reply cpp/binary/main.cc]: done.

File: cpp/binary/main.cc (hunk 2 above)
Comment (terryheo) [reply cpp/binary/main.cc]: To me, it sounds like a side effect. Why don't you remove the side effect? We might compare string with case insensitive.

Comment (thaink) [reply cpp/binary/main.cc]: Done. Using string_view now.

**Pull request merged**